fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/storestate/ReadVersionRecordStoreStateCacheFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * ReadVersionRecordStoreInfoCacheFactory.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.storestate;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabase;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A factory for creating {@link ReadVersionRecordStoreStateCache}s.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class ReadVersionRecordStoreStateCacheFactory implements FDBRecordStoreStateCacheFactory {
+    private long refreshTimeMillis;
+    private long deadlineTimeMillis;
+    private long maxSize;
+
+    private ReadVersionRecordStoreStateCacheFactory() {
+        this.refreshTimeMillis = 5000;
```

Comment:
Would you like to add a comment on why 5s? Probably define constants for these (or use already defined constants if they are really derived from some) if it makes sense?

Reply:
Make sense. Thanks.